### ISSUE TITLE
[Crash] [Dopamine 2 preview 5] "Could not find a part of the path"

### ISSUE THREAD

**Issue by Tyf0x** [digimezzo/dopamine-windows#933]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report  
[ ] Feature request
</code></pre>

## I'm using these versions

* Windows version: 10 pro
* Dopamine version: 2.0.0.1005

## Reproduction steps

* Step 1: Add folder to dopamine library
* Step 2: Close dopamine and delete the folder
* Step 3: Open dopamine

## Observed behavior

Dopamine with open then crash right away

## Desired behavior

Dopamine notifies you that it couldn't find a library directory

## Log file

```2018-12-18 11:50:38.949|Error|App.xaml.ExecuteEmergencyStop|537|Unhandled Exception. Exception:
System.IO.DirectoryNotFoundException: Could not find a part of the path '<my-missing-directory-path-here>'.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileSystemEnumerableIterator`1.CommonInit()
   at System.IO.FileSystemEnumerableIterator`1..ctor(String path, String originalUserPath, String searchPattern, SearchOption searchOption, SearchResultHandler`1 resultHandler, Boolean checkHost)
   at System.IO.Directory.GetDirectories(String path)
   at Dopamine.Services.Folders.FoldersService.<>c__DisplayClass14_0.<GetSubfoldersAsync>b__0()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Dopamine.Services.Folders.FoldersService.<GetSubfoldersAsync>d__14.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Dopamine.ViewModels.FullPlayer.Collection.CollectionFoldersViewModel.<GetSubfoldersAsync>d__32.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Dopamine.ViewModels.FullPlayer.Collection.CollectionFoldersViewModel.<FillListsAsync>d__34.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Dopamine.ViewModels.Common.Base.TracksViewModelBase.<LoadedCommandAsync>d__48.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Dopamine.ViewModels.Common.Base.CommonViewModelBase.<<-ctor>b__90_4>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

Stack trace:
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileSystemEnumerableIterator`1.CommonInit()
   at System.IO.FileSystemEnumerableIterator`1..ctor(String path, String originalUserPath, String searchPattern, SearchOption searchOption, SearchResultHandler`1 resultHandler, Boolean checkHost)
   at System.IO.Directory.GetDirectories(String path)
   at Dopamine.Services.Folders.FoldersService.<>c__DisplayClass14_0.<GetSubfoldersAsync>b__0()
   at System.Threading.Tasks.Task.Execute()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Dopamine.Services.Folders.FoldersService.<GetSubfoldersAsync>d__14.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Dopamine.ViewModels.FullPlayer.Collection.CollectionFoldersViewModel.<GetSubfoldersAsync>d__32.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Dopamine.ViewModels.FullPlayer.Collection.CollectionFoldersViewModel.<FillListsAsync>d__34.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Dopamine.ViewModels.Common.Base.TracksViewModelBase.<LoadedCommandAsync>d__48.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Dopamine.ViewModels.Common.Base.CommonViewModelBase.<<-ctor>b__90_4>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

2018-12-18 11:50:38.949|Info|App.xaml.ExecuteEmergencyStop|540|### FORCED STOP of Dopamine, version 2.0.0.1005 ###
2018-12-18 11:50:39.109|Info|Shell.xaml.Window_Closed|326|### STOPPED Dopamine, version 2.0.0.1005 ###```

**Comment by digimezzo** [comment]
@Tyf0x Could you check these things for me:

1. Does the directory exist?
2. If yes, Does Dopamine have permission to read from that directory?

These questions might help me set up a similar situation as yours, to avoid a crash in such cases.

**Comment by Tyf0x** [comment]
Hi @digimezzo,

no, this particular issue happened after I copied the portable version to a new computer missing the directory (so the directory was missing).

**Comment by digimezzo** [comment]
@Tyf0x Thanks for the info. I'll set up a similar situation and make sure this doesn't cause a crash in the next release.

**Comment by digimezzo** [comment]
@Tyf0x This seems to be caused by the Folders screen. I've fixed the bug and will release a new preview tonight which contains the fix.

**Comment by digimezzo** [comment]
Fixed by 2260c04fbd0eca2e029874d0fef54a2c95c9e4bd
